ee/desktop/server/server.go

```diff
@@ -0,0 +1,151 @@
+// server is a http server that listens to a unix socket or named pipe for windows.
+// Its implementation was driven by the need for "launcher proper" to be able to
+// communicate with launcher desktop running as a separate process.
+package server
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+)
+
+type DesktopServer struct {
+	logger       log.Logger
+	server       *http.Server
+	listener     net.Listener
+	shutdownChan chan<- struct{}
+	authToken    string
+	socketPath   string
+}
+
+func New(logger log.Logger, authToken string, socketPath string, shutdownChan chan<- struct{}) (*DesktopServer, error) {
+	desktopServer := &DesktopServer{
+		shutdownChan: shutdownChan,
+		authToken:    authToken,
+		logger:       log.With(logger, "component", "desktop_server"),
+		socketPath:   socketPath,
+	}
+
+	authedMux := http.NewServeMux()
+	authedMux.HandleFunc("/shutdown", desktopServer.shutdownHandler)
+
+	mux := http.NewServeMux()
+	mux.Handle("/", desktopServer.authMiddleware(authedMux))
```

Comment:
Does this need a default route? I forget what mux does as a catch all

Reply:
ahh cool, can pass authed mux as handler wrapped in middleware